src/transaction/transaction_single_node_group_complete_manager.cpp

```diff
@@ -0,0 +1,189 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+//
+// Manager of completed group on single node
+//
+
+#include "log_manager.h"
+#include "thread_manager.hpp"
+#include "transaction_single_node_group_complete_manager.hpp"
+
+namespace cubtx
+{
+  single_node_group_complete_manager::~single_node_group_complete_manager ()
+  {
+  }
+
+  //
+  // get global single node instance
+  //
+  single_node_group_complete_manager *single_node_group_complete_manager::get_instance ()
+  {
+    if (gl_single_node_group == NULL)
+      {
+	gl_single_node_group = new single_node_group_complete_manager ();
+      }
+    return gl_single_node_group;
+  }
+
+  //
+  // init initialize single node group commit
+  //
+  void single_node_group_complete_manager::init ()
+  {
+    cubthread::looper looper = cubthread::looper (std::chrono::milliseconds (10));
+    gl_single_node_group = get_instance ();
+    LSA_SET_NULL (&gl_single_node_group->m_latest_closed_group_log_lsa);
+
+    single_node_group_complete_manager::gl_single_node_group_complete_daemon = cubthread::get_manager ()->create_daemon ((
+		looper),
+	new single_node_group_complete_task (), "single_node_group_complete_daemon");
+  }
+
+  //
+  // final finalizes single node group commit
+  //
+  void single_node_group_complete_manager::final ()
+  {
+    if (gl_single_node_group_complete_daemon != NULL)
+      {
+	cubthread::get_manager()->destroy_daemon (gl_single_node_group_complete_daemon);
+	gl_single_node_group_complete_daemon = NULL;
+      }
+
+    delete gl_single_node_group;
+    gl_single_node_group = NULL;
+  }
+
+  //
+  // notify_log_flush notifies single node group commit.
+  //
+  void single_node_group_complete_manager::notify_log_flush_lsa (const LOG_LSA *lsa)
+  {
+    /* TO DO - disable it temporary since it is not tested */
+    return;
+
+    assert (lsa != NULL && LSA_GE (lsa, &m_latest_closed_group_log_lsa));
+
+    /* TODO - use m_latest_closed_group_stream_start_positon, m_latest_closed_group_stream_end_positon */
+    if (LSA_GT (lsa, &m_latest_closed_group_log_lsa))
+      {
+	cubthread::entry *thread_p = &cubthread::get_entry();
+	do_complete (thread_p);
+      }
+  }
+
+  //
+  // on_register_transaction - on register transaction specific to single node.
+  //
+  void single_node_group_complete_manager::on_register_transaction ()
+  {
+    /* This function is called after adding a transaction to the current group.
+     * Currently, we wakeup GC thread when first transaction is added into current group.
+     */
+    if (is_latest_closed_group_completed ()
+	&& get_current_group ().get_container ().size () == 1)
```

Comment:
Even as temporary code, it would be safer to use next code:
```suggestion
	&& get_current_group ().get_container ().size () >= 1)
```

Reply:
As I mentioned in 22945, it is better to not use it if we don't need.